aten/src/ATen/nn.yaml

```diff
@@ -2,91 +2,142 @@
 
 - name: binary_cross_entropy(Tensor self, Tensor target, Tensor weight={}, bool size_average=true, bool reduce=true)
   cname: BCECriterion
+  scalar_check:
+    output: 'true'
 
 - name: kl_div(Tensor self, Tensor target, bool size_average=true, bool reduce=true)
   cname: DistKLDivCriterion
+  scalar_check:
+    output: 'true'
 
 - name: l1_loss(Tensor self, Tensor target, bool size_average=true, bool reduce=True)
   cname: AbsCriterion
+  scalar_check:
+    output: 'true'
 
 - name: mse_loss(Tensor self, Tensor target, bool size_average=true, bool reduce=true)
   cname: MSECriterion
+  scalar_check:
+    output: 'true'
 
 - name: multi_margin_loss(Tensor self, LongTensor target, Scalar p=1, Scalar margin=1, Tensor weight={}, bool size_average=true)
   cname: MultiMarginCriterion
+  scalar_check:
+    output: 'true'
 
 - name: multilabel_margin_loss(Tensor self, LongTensor target, bool size_average=true)
   cname: MultiLabelMarginCriterion
   buffers: [is_target]
+  scalar_check:
+    output: 'true'
+    is_target: target_->isScalar()
 
 - name: nll_loss(Tensor self, LongTensor target, Tensor weight={}, bool size_average=true, int64_t ignore_index=-100, bool reduce=True)
   cname: ClassNLLCriterion
   buffers: [total_weight]
+  scalar_check:
+    output: 'true'
+    total_weight: 'true'
 
 - name: nll_loss2d(Tensor self, LongTensor target, Tensor weight={}, bool size_average=true, int64_t ignore_index=-100, bool reduce=True)
   cname: SpatialClassNLLCriterion
   buffers: [total_weight]
+  scalar_check:
+    output: 'true'
```

Comment:
If this says output is always a scalar then it's wrong, because it's controlled by the `reduce` argument

Reply:
Ah ya, forgot we added that, let me fix.

Reply:
should be fixed now.